protocols/luci-proto-3g/htdocs/luci-static/resources/protocol/3g.js

```diff
@@ -23,15 +23,17 @@ network.registerPatternVirtual(/^3g-.+$/);
 function write_keepalive(section_id, value) {
 	var f_opt = this.map.lookupOption('_keepalive_failure', section_id),
 	    i_opt = this.map.lookupOption('_keepalive_interval', section_id),
-	    f = (f_opt != null) ? +f_opt[0].formvalue(section_id) : null,
-	    i = (i_opt != null) ? +i_opt[0].formvalue(section_id) : null;
-
-	if (f === '' || isNaN(f))
-		f = null;
+	    f_str = (f_opt != null) ? f_opt[0].formvalue(section_id) : null,
+	    i_str = (i_opt != null) ? i_opt[0].formvalue(section_id) : null,
+	    f = (f_str === null || f_str === '') ? f_str : +f_str,
+	    i = (i_str === null || i_str === '') ? i_str : +i_str;
 
 	if (i == null || i == '' || isNaN(i) || i < 1)
 		i = 1;
 
+	if (f === null || f === '' || isNaN(f))
+		f = (i == 1) ? null : 5;
+
 	if (f !== null)
 		uci.set('network', section_id, 'keepalive', '%d %d'.format(f, i));
 	else
```

Comment:
Perhaps optional chaining to avoid all the null checks:
```js
function write_keepalive(section_id, value) {
	const f_opt = this.map.lookupOption('_keepalive_failure', section_id)?.[0]?.formvalue(section_id);
	const i_opt = this.map.lookupOption('_keepalive_interval', section_id)?.[0]?.formvalue(section_id);

	let i = parseInt(i_opt, 10) || 1;
	let f = parseInt(f_opt, 10);

	if (isNaN(f)) {
		f = (i === 1) ? null : 5;
	}

	if (f !== null) {
		uci.set('network', section_id, 'keepalive', `${f} ${i}`);
	} else {
		uci.unset('network', section_id, 'keepalive');
	}
}
```

Please test :)

Reply:
Thanks, updated!

Reply:
Doesn't look like it :)

Reply:
OK, I ended up doing a somewhat more conservative change but the "messy" checks for `null` and `''` are nonetheless all gone. Note that I am fully aware of the merits of `let` and `const` over `var` and so on but it is not the only instance thereof and I do not feel like redoing the whole file. I mostly just wanted to fix the regressions and get things working. It is currently in a broken state.

Reply:
OK good. Looks fine. That's fine as you've done it, though we encourage new code to use ES6 or better.